test/ruby_lsp_rails/code_lens_test.rb

```diff
@@ -288,12 +288,29 @@ def change
         assert_match("#{ruby} bin/rails db:migrate VERSION=123456", response[0].command.arguments[0])
       end
 
+      test "recognizes controller actions" do
+        response = generate_code_lens_for_source(<<~RUBY)
+          class UsersController < ApplicationController
+            def index
+            end
+          end
+        RUBY
+        path, line = response[0].command.arguments.first
+
+        assert_equal(1, response.size)
+        assert_match("GET /users(.:format)", response[0].command.title)
```

Comment:
`(.:format)` is an optional parameter. This way, the route can accept `/users` (no format), `/users.json`, and `/users.xml` and define behaviour based on the format provided. See https://guides.rubyonrails.org/routing.html#request-based-constraints, https://guides.rubyonrails.org/routing.html#defining-defaults

I left it as-is because otherwise the developer may not know it is there, and it matches what `bin/rails routes` outputs, but I'm happy to remove it if you think it isn't necessary.

Reply:
Makes sense 👍